omnigibson/utils/object_state_utils.py

```diff
@@ -252,97 +252,79 @@ def sample_kinematics(
     return success
 
 
-# Folded / Unfolded related utils
-m.DEBUG_CLOTH_PROJ_VIS = False
-# Angle threshold for checking smoothness of the cloth; surface normals need to be close enough to the z-axis
-m.NORMAL_Z_ANGLE_DIFF = np.deg2rad(45.0)
-
-
-def calculate_projection_area_and_diagonal_maximum(obj):
+def sample_cloth_on_rigid(obj, other, max_trials=10, z_offset=0.05, randomize_xy=True):
     """
-    Calculate the maximum projection area and the diagonal length along different axes
+    Samples the cloth object @obj on the rigid object @other
 
     Args:
-        obj (DatasetObject): Must be PrimType.CLOTH
+        obj (StatefulObject): Object whose state should be sampled. e.g.: for sampling a bed sheet on a rack,
+            @obj is the bed sheet
+        other (StatefulObject): Object who is the reference point for @obj's state. e.g.: for sampling a bed sheet
+            on a rack, @other is the rack
+        max_trials (int): Number of attempts for sampling
+        z_offset (float): Z-offset to apply to the sampled pose
 
     Returns:
-        area_max (float): area of the convex hull of the projected points
-        diagonal_max (float): diagonal of the convex hull of the projected points
+        bool: True if successfully sampled, else False
     """
-    # use the largest projection area as the unfolded area
-    area_max = 0.0
-    diagonal_max = 0.0
-    dims_list = [[0, 1], [0, 2], [1, 2]]  # x-y plane, x-z plane, y-z plane
+    assert z_offset > 0.5 * 9.81 * (og.sim.get_physics_dt() ** 2) + 0.02,\
+        f"z_offset {z_offset} is too small for the current physics_dt {og.sim.get_physics_dt()}"
 
-    for dims in dims_list:
-        area, diagonal = calculate_projection_area_and_diagonal(obj, dims)
-        if area > area_max:
-            area_max = area
-            diagonal_max = diagonal
+    if not (obj.prim_type == PrimType.CLOTH and other.prim_type == PrimType.RIGID):
+        raise ValueError("sample_cloth_on_rigid requires obj1 is cloth and obj2 is rigid.")
 
-    return area_max, diagonal_max
+    state = og.sim.dump_state(serialized=False)
 
+    # Reset the cloth
+    obj.root_link.reset()
 
-def calculate_projection_area_and_diagonal(obj, dims):
-    """
-    Calculate the projection area and the diagonal length when projecting to the plane defined by the input dims
-    E.g. if dims is [0, 1], the points will be projected onto the x-y plane.
+    obj_aabb_low, obj_aabb_high = obj.states[AABB].get_value()
```

Comment:
are these values correct? what if they're already queried before `obj.root_link.reset()` was called? maybe we should clear the cache of these values first just to be safe